LayoutTests/imported/w3c/web-platform-tests/resources/testdriver-vendor.js

```diff
@@ -289,89 +308,97 @@ window.test_driver_internal.action_sequence = async function(sources)
         await dispatchMouseActions(pointerSource.actions, pointerType);
 };
 
-window.test_driver_internal.set_permission = async function(permission_params, context=null)
+/**
+ *
+ * @param {Object} permission_params
+ * @param {PermissionDescriptor} permission_params.descriptor
+ * @param {PermissionState} permission_params.state
+ * @returns Promise<undefined>
+ */
+window.test_driver_internal.set_permission = async function(permission_params)
 {
     switch (permission_params.descriptor.name) {
     case "geolocation":
-        if (!window.testRunner?.setGeolocationPermission)
-            throw new Error("unimplemented");
-        setInterval(() => {
-            window.testRunner.setMockGeolocationPosition(51.478, -0.166, 100);
-        }, 100);
+        await pause(100);
```

Comment:
setInterval != setTimeout :)

Reply:
ah yeah... seemed weird that it kept setting it to the same position but I guess it's just reporting a value. 

Reminds me... we should maybe add something Web Driver for testing Geolocation.